packages/core/test/render3/directive_spec.ts

```diff
@@ -121,6 +122,104 @@ describe('directive', () => {
          expect(directiveInstance !.testValue).toBe(false);
        });
 
+    it('should match directives on <ng-template>', () => {
+      /**
+       *   @Directive({
+       *     selector: 'ng-template[directiveA]'
+       *   })
+       *   export class DirectiveA {
+       *     constructor(public templateRef: TemplateRef<any>) {}
+       *   }
+       */
+      let tmplRef: any;
+      class DirectiveA {
+        constructor(public templateRef: any) { tmplRef = templateRef; }
+        static ngDirectiveDef = defineDirective({
+          type: DirectiveA,
+          selectors: [['ng-template', 'directiveA', '']],
+          factory: () => new DirectiveA(directiveInject(TemplateRef as any))
+        });
+      }
+
+      function MyComponent_ng_template_Template_0(rf: RenderFlags, ctx: any) {
+        if (rf & 1) {
+          text(0, 'Some content');
+        }
+      }
+      class MyComponent {
+        static ngComponentDef = defineComponent({
+          type: MyComponent,
+          selectors: [['my-component']],
+          factory: () => new MyComponent(),
+          consts: 2,
+          vars: 0,
+          // <ng-template directiveA>Some content</ng-template>
+          template: function MyComponent_Template(rf: RenderFlags, ctx: any) {
+            if (rf & RenderFlags.Create) {
+              template(
+                  0, MyComponent_ng_template_Template_0, 1, 0, 'ng-template', ['directiveA', '']);
+            }
+          },
+          directives: [DirectiveA, NgIf]
+        });
+      }
+
+      new ComponentFixture(MyComponent);
+      expect(tmplRef instanceof TemplateRef).toBeTruthy();
+    });
+
+    it('should match directives on <ng-container>', () => {
+      /**
+       *   @Directive({
+       *     selector: 'ng-container[directiveA]'
+       *   })
+       *   export class DirectiveA {
+       *     constructor(public vcRef: ViewContainerRef<any>) {}
+       *   }
+       */
+      let vcRef: any;
+      class DirectiveA {
+        constructor(public viewContainerRef: any) { vcRef = viewContainerRef; }
+        static ngDirectiveDef = defineDirective({
+          type: DirectiveA,
+          selectors: [['ng-container', 'directiveA', '']],
+          factory: () => new DirectiveA(directiveInject(ViewContainerRef as any))
+        });
+      }
+
+      function MyComponent_ng_container_Template_0(rf: RenderFlags, ctx: any) {
+        if (rf & 1) {
```

Comment:
Nit: Can we use `RenderFlags` in tests for clarity? esp in case we change the enum.

```suggestion
        if (rf & RenderFlags.Create) {
```